dts/bindings/mfd/infineon,tle9104.yaml

```diff
@@ -0,0 +1,72 @@
+#
+# Copyright (c) 2024 SILA Embedded Solutions GmbH
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+description: Infineon TLE9104 4-channel powertrain switch
+
+compatible: "infineon,tle9104"
+
+include: spi-device.yaml
+
+bus: tle9104
+
+properties:
+  en-gpios:
+    type: phandle-array
+    description: "GPIO for enable"
+
+  resn-gpios:
+    type: phandle-array
+    description: "GPIO for reset"
+
+  in1-gpios:
+    type: phandle-array
+    description: "GPIO for controlling OUT1"
+
+  in2-gpios:
+    type: phandle-array
+    description: "GPIO for controlling OUT2"
+
+  in3-gpios:
+    type: phandle-array
+    description: "GPIO for controlling OUT3"
+
+  in4-gpios:
+    type: phandle-array
+    description: "GPIO for controlling OUT4"
+
+  diagnostic-filter-time:
+    type: int
+    default: 500
+    enum:
+      - 2000
+      - 1000
+      - 500
+      - 200
+    description: "diagnostic filter time in us"
```

Comment:
Default values need justification (e.g. `default matches the power-on reset value` or similar)

Reply:
done